app/services/ml.py

```diff
@@ -1,11 +1,92 @@
-import random
+from datetime import datetime, timedelta
+from typing import List
+
+import pandas as pd
+import tweepy
+from fastapi import Depends
+
+import model
+
+from .scrape import TwitterID, TwitterScraper
+
+
+def _find_replies_in_conversation(tweet_id: int, conversation: List[tweepy.Tweet]):
+    replies = []
+    for tweet in conversation:
+        for reference_tweet in tweet.referenced_tweets:
+            if reference_tweet.type == "replied_to" and reference_tweet.id == tweet_id:
+                replies.append(
+                    {
+                        "id": tweet.id,
+                        "text": tweet.text,
+                        "parent_id": tweet_id,
+                    }
+                )
+                break
```

Comment:
Does this means that one tweet can reference to many? and here we're checking if the parent tweet exists in all the referenced ones?

Reply:
@phhai-hcmut help me ping ping

Reply:
This is for sanity check